dali/pipeline/operator/builtin/external_source.h

```diff
@@ -287,6 +297,23 @@ class ExternalSource : public Operator<Backend>, virtual public BatchSizeProvide
   DISABLE_COPY_MOVE_ASSIGN(ExternalSource);
 
  protected:
+  inline bool HasNdim() {
```

Comment:
What's the reason for making those `inline`? Per [standard specification](https://en.cppreference.com/w/cpp/language/inline)
> A function defined entirely inside a [class/struct/union definition](https://en.cppreference.com/w/cpp/language/classes), whether it's a member function or a non-member friend function, is implicitly an inline function

Reply:
Every other method in this class had `inline`, so I followed. Yup, it doesn't make sense

Reply:
removed